coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -292,22 +298,16 @@ private SqlNode visitLateralViewExplode(List<SqlNode> sqlNodes, List<SqlNode> al
       unnestOperand = hiveIfFunction.createCall(SqlLiteral.createCharString("if", ZERO),
           ImmutableList.of(ifCondition, unnestOperand, arrayOrMapOfNull), null);
     }
-    if (operandCount == 3) { // unnest explode array
-      unnestCall = HiveExplodeOperator.EXPLODE.createCall(ZERO,
-          SqlStdOperatorTable.AS.createCall(ZERO, unnestOperand, aliasOperands.get(2)));
-      unnestCall = SqlStdOperatorTable.AS.createCall(ZERO, unnestCall, aliasOperands.get(1), aliasOperands.get(2));
-    } else { // unnest explode map
-      unnestCall = HiveExplodeOperator.EXPLODE.createCall(ZERO, unnestOperand);
-    }
-    SqlNode rightSelect = new SqlSelect(ZERO, null, new SqlNodeList(ImmutableList.of(SqlIdentifier.star(ZERO)), ZERO),
-        unnestCall, null, null, null, null, null, null, null);
-    SqlNode lateralCall = SqlStdOperatorTable.LATERAL.createCall(ZERO, rightSelect);
-    List<SqlNode> aliasCallOperands = new ArrayList<>();
-    aliasCallOperands.add(lateralCall);
-    aliasCallOperands.addAll(aliasOperands.subList(1, operandCount));
-    SqlCall aliasCall = SqlStdOperatorTable.AS.createCall(ZERO, aliasCallOperands);
+    unnestCall = HiveExplodeOperator.EXPLODE.createCall(ZERO, unnestOperand);
+
+    List<SqlNode> asOperands = new ArrayList<>();
+    asOperands.add(unnestCall);
+    asOperands.addAll(aliasOperands.subList(1, aliasOperands.size()));
+    SqlNode as = SqlStdOperatorTable.AS.createCall(ZERO, asOperands);
+
+    SqlNode lateralCall = SqlStdOperatorTable.LATERAL.createCall(ZERO, as);
```

Comment:
Could you add a comment to the code explaining how this generically handles all cases of array, map, and array/map with auto derived aliases? (just by explaining the mapping of the operators created here to `op0... op3` and their meaning in different cases)

Reply:
Added in the next commit.